assets/src/scripts/monitoring-location/components/hydrograph/download-data.test.js

```diff
@@ -0,0 +1,131 @@
+        });
+
+        afterEach(() => {
+            div.remove();
+        });
+
+        it('Renders form with the appropriate checkboxes and download button', () => {
+            expect(div.selectAll('input[type="checkbox"]').size()).toBe(3);
+            expect(div.selectAll('input[value="primary"]').size()).toBe(1);
+            expect(div.selectAll('input[value="groundwater-levels"]').size()).toBe(1);
+            expect(div.selectAll('input[value="site"]').size()).toBe(1);
+            expect(div.selectAll('button.download-selected-data').size()).toBe(1);
+        });
+
+        it('Rerenders the checkboxes if data visibility changes', () => {
+            store.dispatch(setCompareDataVisibility(true));
+            store.dispatch(setMedianDataVisibility(true));
+            return new Promise(resolve => {
+                window.requestAnimationFrame(() => {
+                    expect(div.selectAll('input[type="checkbox"]').size()).toBe(5);
+                    expect(div.selectAll('input[value="compare"]').size()).toBe(1);
+                    expect(div.selectAll('input[value="median"]').size()).toBe(1);
+                    resolve();
+                });
+            });
+        });
+
+        it('Shows an error message if the download button is clicked with no checkboxes checked', () => {
+            const downloadButton = div.select('button.download-selected-data');
+            downloadButton.dispatch('click');
+            expect(div.select('.usa-alert--error').size()).toBe(1);
+        });
+
+        it('Opens a window with the URL for the selected data', () => {
+            const downloadButton = div.select('button.download-selected-data');
+            div.select('input[value="site"]').property('checked', true);
+            downloadButton.dispatch('click');
+
+            expect(div.select('.usa-alert--error').size()).toBe(0);
+            expect(windowSpy.mock.calls).toHaveLength(1);
+            expect(windowSpy.mock.calls[0][0]).toContain('/site/');
+            expect(windowSpy.mock.calls[0][0]).toContain('sites=11112222');
+
+        });
+
+        it('Opens a window for each selected data', () => {
+            const downloadButton = div.select('button.download-selected-data');
+            store.dispatch(setMedianDataVisibility(true));
+            return new Promise(resolve => {
+                window.requestAnimationFrame(() => {
+                    div.select('input[value="primary"]').property('checked', true);
+                    div.select('input[value="groundwater-levels"]').property('checked', true);
+                    div.select('input[value="median"]').property('checked', true);
+                    downloadButton.dispatch('click');
+
+                    expect(windowSpy.mock.calls).toHaveLength(3);
+
+                    expect(windowSpy.mock.calls[0][0]).toContain('/iv/');
+                    expect(windowSpy.mock.calls[0][0]).toContain('sites=11112222');
```

Comment:
I write the tests like this because the order of the parameters in the URL is meaningless and can change. Doing it this way also has the benefit of pinpointing what the issue.